src/new_currencies.jl

```diff
@@ -0,0 +1,61 @@
+abstract type Currency end
+
+struct Asset{C}
+    balance::Number
+    function Asset{C}(a::Number) where C <: Currency
+        C()
+        new{C}(a)
+    end
+end
+
+
+struct RegularCurrency{S} <: Currency
+    function RegularCurrency{S}() where {S}
+        haskey(_regular_data, S) || error("Regular currency $S is not defined.")
+        new{S}()
+    end
+end
+
+const _regular_data = Dict(
+	:DAI  => (RegularCurrency{:DAI},  18),
+	:AAVE => (RegularCurrency{:AAVE}, 18))
+
+struct aToken{S, P} <: Currency
+    function aToken{S, P}() where {S, P}
+        haskey(_atoken_data, S) || error("aToken $S is not defined.")
+        new{S, P}()
+    end
+end
+
+const _atoken_data = Dict(
+	:aDAI  => (aToken{:DAI},  :DAI, RegularCurrency{:DAI}),
```

Comment:
Are all 3 values required?

Reply:
The third one is not necessary. We can resolve it this way:
```suggestion
	:aDAI  => (aToken{:DAI},  :DAI),
	peg(a::Asset{aToken{S, T}}) where {S, T} = Asset{RegularCurrency{T}}
```